src/SFML/Graphics/Shape.cpp

```diff
@@ -221,8 +221,8 @@ void Shape::update()
     m_insideBounds = m_vertices.getBounds();
 
     // Compute the center and make it the first vertex
-    m_vertices[0].position.x = m_insideBounds.left + m_insideBounds.width / 2;
-    m_vertices[0].position.y = m_insideBounds.top + m_insideBounds.height / 2;
+    m_vertices[0].position = Vector2f(m_insideBounds.left, m_insideBounds.top) +
+                             Vector2f(m_insideBounds.width, m_insideBounds.height) / 2.f;
```

Comment:
We should really think about `Rect::getEnd()` and `Rect::getCenter()` methods.

This would be:
```suggestion
    m_vertices[0].position = m_insideBounds.getCenter();
```

But at least, we can already do now -- which is of course much less expressive:

```suggestion
    m_vertices[0].position = m_insideBounds.getPosition() + m_insideBounds.getSize() / 2.f;
```

Reply:
I agree that having named APIs for these other parts of a rectangle would be useful. I support their addition.